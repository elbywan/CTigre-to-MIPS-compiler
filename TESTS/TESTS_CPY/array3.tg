/* This program test arrays of arrays */
type intArray = array of int in
type intMatrix = array of intArray in

var N := 3
and x := 100 in
var matrix := intMatrix[N] of (intArray[N] of x:=x+1; x) in

for i=0 to N-1 do
  for j=0 to N-1 do
    print("[");printint(i);print("][");printint(j);print("] = ");printint(matrix[i][j]);print("\n")
  done
done
