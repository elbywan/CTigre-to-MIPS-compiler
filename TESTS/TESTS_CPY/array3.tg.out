[0][0] = 101
[0][1] = 102
[0][2] = 103
[1][0] = 104
[1][1] = 105
[1][2] = 106
[2][0] = 107
[2][1] = 108
[2][2] = 109

